gtsam/discrete/DiscreteValues.cpp

```diff
@@ -26,12 +26,24 @@ using std::stringstream;
 
 namespace gtsam {
 
+/* ************************************************************************ */
+static void stream(std::ostream& os, const DiscreteValues& x,
+                   const KeyFormatter& keyFormatter) {
+  for (const auto& kv : x)
+    os << "(" << keyFormatter(kv.first) << ", " << kv.second << ")";
+}
+
+/* ************************************************************************ */
+std::ostream& operator<<(std::ostream& os, const DiscreteValues& x) {
```

Comment:
I like this. I did this for a bunch of classes in `geometry`, really helps with flexibility in using both `.print()` and `<<`.

We could make this a potential GSoC project.